src/model/points-model.js

```diff
@@ -1,10 +1,51 @@
-import { getRandomPoint } from '../mock/points';
+import Observable from '../framework/observable.js';
+import { getRandomPoint } from '../mock/points.js';
 
 const POINT_COUNT = 6;
-export default class PointsModel {
+export default class PointsModel extends Observable {
   #points = Array.from({length: POINT_COUNT}, getRandomPoint);
 
   get points() {
     return this.#points;
   }
+
+  updatePoint(updateType, update) {
+    const index = this.#points.findIndex((point) => point.id === update.id);
+
+    if (index === -1) {
+      throw new Error('Can\'t update unexisting point');
+    }
+
```

Comment:
this.#points = this.#points.map((point) => point.id === update.id ? update : point )